stellargraph/datasets/datasets.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019-2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Sample datasets for stellargraph demonstrations
+"""
+
+from .dataset_loader import DatasetLoader
+from typing import Optional
+
+
+def download_all_datasets(ignore_cache: Optional[bool] = False) -> None:
+    """ download all datasets """
+    # obtain all the classes derived from DatasetLoader (note: must be direct descendants)
+    for cls in DatasetLoader.__subclasses__():
+        dataset = cls()
+        dataset.download(ignore_cache)
+
+
+class Cora(DatasetLoader):
+    def __init__(self) -> None:
+        super().__init__(
+            "Cora",
+            "cora",
+            "https://linqs-data.soe.ucsc.edu/public/lbc/cora.tgz",
+            url_archive_format="gztar",
```

Comment:
I suspect all of these can be inferred from the URL itself; is it worth repeating this information? Maybe we could have `url_archive_format` allow `None` to mean "infer", so we don't have to specify it for "obvious" cases like this. That said, optimising this is probably low-value.

Reply:
I thought about this, and it seemed more complexity for little gain (either attempting to parse and map the end of the URL, or the http request headers)